lib/src/components/NxModal/NxModal.tsx

```diff
@@ -4,147 +4,40 @@
-     * the browser around modals to simplify the NxModal styling around z-index handling
-     */
-    if (hasNativeModalSupport) {
-      // https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1029#issuecomment-968299542
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      (el as any).showModal();
-    }
-    else {
-      // without native support we don't trap focus in the modal, but we can at least start it off there
-      el.focus();
-    }
-
-    return () => {
-      if (previouslyFocusedEl && previouslyFocusedEl instanceof HTMLElement) {
-        // The useEffect cleanup executes while the modal is still present (in React 16 at least). While the modal
-        // still exists, the document is still "blocked by the modal dialog" so trying to focus elements outside of
-        // it won't work. So we have to wait until the next cycle of the event loop when it's gone
-        Promise.resolve().then(() => previouslyFocusedEl.focus());
-      }
-    };
-  }, []);
-
-  // listen to the native HTMLDialogElement cancel event which supporting browsers fire when the modal is closed
-  // via ESC
-  useEffect(function() {
-    const dialog = dialogRef.current;
-
-    if (dialog && hasNativeModalSupport) {
-      /* eslint-disable @typescript-eslint/no-non-null-assertion */
-      dialog.addEventListener('cancel', onCancel!);
-      return () => { dialog.removeEventListener('cancel', onCancel!); };
-      /* eslint-enable @typescript-eslint/no-non-null-assertion */
-    }
-    else {
-      return undefined;
-    }
-  }, [onCancel]);
+    'nx-modal--wide': variant === 'wide',
+    'nx-modal--narrow': variant === 'narrow'
+  });
 
   const ariaLabelAttrNames = ['aria-label', 'aria-labelledby'] as const,
       ariaLabels = pick(ariaLabelAttrNames, attrs),
       attrsWithoutLabels = omit(ariaLabelAttrNames, attrs);
 
   return (
-    // Provide the dialog element to descendants so that tooltips can attach to it instead of the body,
-    // which is necessary so that they end up in the top layer rather than behind the modal
-    <NxModalContext.Provider value={dialogRefState}>
-      {/* Note: role="dialog" should be redundant but I think some screenreaders (ChromeVox) don't know
-        * what a <dialog> is.  It makes a difference there.
-        *
-        * The tabindex is for the sake of browsers which don't support dialog. In those browsers we try to
-        * focus the dialog element itself when it opens which can't be done if it doesn't have a tab index
-        */}
-      <dialog ref={dialogRef}
-              role={role || 'dialog'}
-              aria-modal="true"
-              className="nx-modal-backdrop"
-              tabIndex={-1}
-              onKeyDown={dialogKeydownListener}
-              {...ariaLabels}>
-        <div className={modalClasses} {...attrsWithoutLabels}/>
-      </dialog>
-    </NxModalContext.Provider>
+    <AbstractDialog role={role || 'dialog'}
```

Comment:
This default is already in `AbstractDialog`, no need to repeat it
```suggestion
    <AbstractDialog role={role}
```